src/storage/mapstorage/ResourceStoreMapStorage.ts

```diff
@@ -0,0 +1,61 @@
+import { BasicRepresentation } from '../../ldp/representation/BasicRepresentation';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
+import { readableToString } from '../../util/StreamUtil';
+import type { ResourceStore } from '../ResourceStore';
+import type { MapStorage } from './MapStorage';
+
+/**
+ * A {@link MapStorage} for strings using a {@link ResourceStore} as backend.
+ *
+ * Values will be sent as data streams to the given identifiers,
+ * so how these are stored depend on the underlying store.
+ *
+ * All non-404 errors will be re-thrown.
+ */
+export class ResourceStoreMapStorage implements MapStorage<string> {
+  private readonly source: ResourceStore;
+
+  public constructor(source: ResourceStore) {
+    this.source = source;
+  }
+
+  public async get(identifier: ResourceIdentifier): Promise<string | undefined> {
+    try {
+      const representation = await this.source.getRepresentation(identifier, { type: { 'text/plain': 1 }});
```

Comment:
True. I was thinking of just having this storage be one that simply returns a buffer (typed as string here) which could then be wrapped around by other storages for specific types. E.g., could also have one that returns Quads by parsing this string through a turtle parser. But I guess we can still do this if we enforce JSON values and such a storage then stores its RDF string in a single JSON field. One disadvantage is that we can't really type that we accept JSON (afaik?), so that might give the impression that any type of object could be stored here.